src/devices/cpu/gigatron/gigatron.cpp

```diff
@@ -65,16 +86,41 @@ void gigatron_cpu_device::device_start()
 	m_program = &space(AS_PROGRAM);
 	m_data = &space(AS_DATA);
 
-	save_item(NAME(m_pc));
-	save_item(NAME(m_flags));
+	init();
+}
+
+void gigatron_cpu_device::init()
+{
+	ac = 0;
+	x = 0;
+	y = 0;
+	m_pc = 0;
+	state_add(GTRON_A,         "AC",         ac);
+	state_add(GTRON_X,         "X",         x);
+	state_add(GTRON_Y,         "Y",         y);
+}
+
+void gigatron_cpu_device::branchOp(int op, int mode, int bus, int d)
+{
+}
 
-	// Register state for debugger
-	state_add( GTRON_R0, "PC", m_pc ).formatstr("%02X");
-	state_add( STATE_GENPC, "GENPC", m_r[7] ).noshow();
-	state_add( STATE_GENPCBASE, "CURPC", m_r[7] ).noshow();
-	state_add( STATE_GENFLAGS, "GENFLAGS", m_flags ).noshow();
+void gigatron_cpu_device::aluOp(int op, int mode, int bus, int d)
+{
+	int b;
+	switch(bus) {
+		case 0:
+			b = d;
+			break;
+		case 1:
+		case 2:
+			b = ac;
+			break;
+		case 3:
+	}
+}
 
-	set_icountptr(m_icount);
```

Comment:
This statement should not be deleted. Initializing the icount pointer is an obligatory part of `device_start` for CPUs. Otherwise, MAME's emulation core will throw a "m_icountptr never initialized!" error.

Reply:
Fixed.